libbeat/docs/queueconfig.asciidoc

```diff
@@ -76,3 +77,153 @@ will be immediately available for consumption.
 
 The default values is 0s.
 
+[float]
+[[configuration-internal-queue-spool]]
+=== Configure the file spool queue
+
+The file spool queue stores all events in an on disk ring buffer. The spool
+has a write buffer, which new events are written to. Events written to the
+spool are forwarded to the outputs, only after the write buffer has been
+flushed successfully.
+
+The spool waits for the output to acknowledge or drop events. If the spool is
+full, no new events can be inserted. The spool will block. Space is freed only
+after a signal from the output has been received.
+
+On disk, the spool divides a file into pages. The `file.page_size` setting
+configures the file's page size at file creation time. The optimal page size depends
+on the effective block size, used by the underlying file system.
+
+This sample configuration enables the spool with all default settings (See
+<<configuration-internal-queue-spool-reference>> for defaults) and the
+default file path:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.spool: ~
+------------------------------------------------------------------------------
+
+This sample configuration creates a spool of 512MiB, with 16KiB pages. The
+write buffer is flushed if 10MiB of contents, or 1024 events have been
+written. If the oldest available event has been waiting for 5s in the write
+buffer, the buffer will be flushed as well:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.spool:
+  file:
+    path: "${path.data}/spool.dat"
+    size: 512MiB
+    page_size: 16KiB
+  write:
+    buffer_size: 10MiB
+    flush.timeout: 5s
+    flush.events: 1024
+------------------------------------------------------------------------------
+
+[float]
+[[configuration-internal-queue-spool-reference]]
+==== Configuration options
+
+You can specify the following options in the `queue.spool` section of the
++{beatname_lc}.yml+ config file:
+
+[float]
+===== `file.path`
+
+The spool file path. The file is created on startup, if it does not exist.
+
+The default value is "${path.data}/spool.dat".
+
+===== `file.permissions`
```

Comment:
@urso The content looks good, but just noticed that some of the sections are missing `[float]` tags starting with `file.permissions`.